resources/assets/js/components/JobDetails/JobDetails.tsx

```diff
@@ -0,0 +1,734 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState } from "react";
+import ReactDOM from "react-dom";
+import { Formik, Form, Field } from "formik";
+import * as Yup from "yup";
+import Modal from "../Modal";
+
+const JobSchema = Yup.object().shape({
+  title: Yup.string()
+    .min(2, "Too Short!")
+    .max(50, "Too Long!")
+    .required("This field is required."),
+  termLength: Yup.number()
+    .min(1, "Too Short!")
+    .max(36, "Too Long!")
+    .required("This field is required."),
+  classification: Yup.mixed()
+    .oneOf(
+      ["AS", "BI", "CO", "CR", "CS", "EC", "EX", "FO", "IS", "PC", "PE", "PM"],
+      "Please select from the available options",
+    )
+    .required("This field is required."),
+  level: Yup.number()
+    .min(1, "Levels begin at one")
+    .max(9, "There are only nine levels")
+    .required("This field is required."),
+  securityLevel: Yup.number()
+    .min(1, "")
+    .max(3, "")
+    .required("This field is required."),
+  language: Yup.number()
+    .min(1, "")
+    .max(5, "")
+    .required("This field is required."),
+  city: Yup.string()
+    .min(3, "Too short!")
+    .max(50, "Too long!")
+    .required("This field is required."),
+  province: Yup.mixed()
+    .oneOf(
+      [
+        "AB",
+        "BC",
+        "MB",
+        "NB",
+        "NL",
+        "NS",
+        "NT",
+        "NU",
+        "ON",
+        "PE",
+        "QC",
+        "SK",
+        "YT",
+      ],
+      "Please select from the available options",
+    )
+    .required("This field is required."),
+  remoteWork: Yup.mixed()
+    .oneOf(
+      ["remoteWorkWorld", "remoteWorkCanada", "remoteWorkNone"],
+      "Please select from the available options",
+    )
+    .required("This field is required."),
+  telework: Yup.mixed()
+    .oneOf(
+      [
+        "teleworkAlways",
+        "teleworkFrequently",
+        "teleworkOccasionally",
+        "teleworkSometimes",
+        "teleworkNever",
+      ],
+      "Please select from the available options",
+    )
+    .required("This field is required."),
+  flexHours: Yup.mixed()
+    .oneOf(
+      [
+        "flexHoursAlways",
+        "flexHoursFrequently",
+        "flexHoursOccasionally",
+        "flexHoursSometimes",
+        "flexHoursNever",
+      ],
+      "Please select from the available options",
+    )
+    .required("This field is required."),
+});
+
+const TextInput = ({
```

Comment:
Formik makes use of some extra props, like 
```js
  field: { name, value, onChange, onBlur },
  form: { errors },
```
we can look at making them optional I suppose, I thought it would be easier to have some separate form components

Reply:
Hmm. Yoni did add the field and form props to the Input.tsx component on his branch. But the result is that most props are optional, and a bit confusing. I support having seperate Formik components that wrap the originals.

Reply:
https://github.com/GCTC-NTGC/TalentCloud/blob/feature/job-builder/step-0/resources/assets/js/components/Forms/Input.tsx 

This is what is looks like currently.